### PR TITLE
#199 Complete Announcements Controller

### PR DESCRIPTION
**Pull request by chriswininger**
Corrected behavior and completed logic for allowing editing and display
of announcements in the refactored NestJS API

### CODE REVIEW

File: api-server-nest/src/announcements/announcements.service.ts [hunk 1]
```diff
@@ -22,7 +22,12 @@ export class AnnouncementsService {
     ensureOne(announcement: CreateOrUpdateAnnouncementRequest): Promise<AnnouncementModel> {
         const whereAnyAnnouncementExists = sequelize.literal('true = true')
 
-        return this.announcementModel.findOrCreate({ where: whereAnyAnnouncementExists, defaults: announcement })
+        const id = uuidv4()
+
```
Comment (chriswininger) [inline api-server-nest/src/announcements/announcements.service.ts:26]: ^ got to provide a possible id for the case where the announcement doesn't exist

File: api-server-nest/src/announcements/announcements.service.ts [hunk 2]
```diff
@@ -32,13 +37,14 @@ export class AnnouncementsService {
         return this.announcementModel.create({ ...newAnnouncement, id })
     }
 
-    update(id: string, updatedAnnouncement: CreateOrUpdateAnnouncementRequest): Promise<DbUpdateResponse<AnnouncementModel>> {
+    update(id: string, updatedAnnouncement: CreateOrUpdateAnnouncementRequest): Promise<AnnouncementModel> {
         const updateQueryOptions: UpdateOptions = {
             where: {id},
```
Comment (chriswininger) [inline api-server-nest/src/announcements/announcements.service.ts:42]: ^ refactored the return a bit to make this nicer for the controller

File: api-server-nest/src/announcements/dto/create-or-update-announcement-request.ts [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
         if (isNotNullOrUndefined(copy)) {
             cloneAttributes<CreateOrUpdateAnnouncementRequest>(copy, this)
         }
     }
 
     @ApiProperty({example: 'Life is not a dream. Beware. And beware. And beware.'})
-    @IsNotEmpty()
+    @IsOptional()
     message: string
 }
```
Comment (chriswininger) [inline api-server-nest/src/announcements/dto/create-or-update-announcement-request.ts:16]: We want to allow this to be empty for the initial case where we create a blank announcement.

FYI, I hit a strange bug where once I removed IsNotEmpty but before I added IsOptional the message was not coming through to to my controller method. I'm not sure if that's a bug in the framework or if some type of decorator is necessary for the validatedTransformPipe we are using, something to watch out for.

File: api-server-nest/src/events/dto/create-event-request.ts [hunk 4]
```diff
@@ -1,6 +1,4 @@
-import {Column, ForeignKey} from "sequelize-typescript";
 import {ApiProperty} from "@nestjs/swagger";
-import {VenueModel} from "../../venues/models/venue.model";
 import {StartEndTimePairs} from "../../shared-types/start-end-time-pairs";
 import isNotNullOrUndefined from "../../utils/is-not-null-or-undefined";
 import cloneAttributes from "../../utils/clone-attributes";
```
Comment (chriswininger) [inline api-server-nest/src/events/dto/create-event-request.ts:4]: scouting, lots of no longer used imports

File: api-server-nest/src/events/events.authenticated.controller.ts [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import {Body, Controller, Get, Param, Put, UseGuards, UseInterceptors} from "@nestjs/common";
+import {Body, Controller, Get, Param, Put, UseGuards} from "@nestjs/common";
 import {VERSION_1_URI} from "../utils/versionts";
```
Comment (chriswininger) [inline api-server-nest/src/events/events.authenticated.controller.ts:2]: ^ more person scouting

**Review by MatthewGidcomb — APPROVED** [review]

**Pull request merged**